analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -504,7 +510,18 @@ open class Npm(
             addAll(getModuleInfo(moduleDir, scopes)!!.dependencies)
 
             workspaceModuleDirs.forEach { workspaceModuleDir ->
-                addAll(getModuleInfo(workspaceModuleDir, scopes, listOf(moduleDir))!!.dependencies)
+                val childNodeModulesDir = workspaceModuleDir.resolve("node_modules")
+                if (childNodeModulesDir.isDirectory && managerName == "PNPM") {
```

Comment:
Similar to my other comment about extensibility of `Npm`. If `Npm` provided a `processModuleDir()` function (or similar), `Pnpm` could override it and apply the stashing before calling the super function.